server/index.js

```diff
@@ -123,6 +124,50 @@ app.get('/api/vehicles/:vehicleId/records', (req, res, next) => {
     })
     .catch(err => next(err));
 });
+
+app.put('/api/garage/edit-car/:vehicleId', uploadsMiddleware, (req, res, next) => {
+  const { vehicleId } = req.params;
+  if (vehicleId < 1 || !Number(vehicleId)) {
+    throw new ClientError(400, 'vehicleId must be a positive integer');
+  }
+  const { year, make, model } = req.body;
+  let photoUrl = null;
+  if (req.file) {
+    photoUrl = `/images/${req.file.filename}`;
+  }
+  const params = [year, make, model, photoUrl, vehicleId];
+  const sql = `
+    update  "vehicles"
+       set  "year" = coalesce(nullif($1,'')::integer, (
+         select "year"
+           from "vehicles"
+          where "vehicleId" = $5
+       )),
+            "make" = coalesce(nullif($2, ''), (
+         select "make"
+           from "vehicles"
+          where "vehicleId" = $5
+       )),
+            "model" = coalesce(nullif($3, ''), (
+         select "model"
+           from "vehicles"
+          where "vehicleId" = $5
+       )),
+            "photoUrl" = coalesce(nullif($4, ''), (
+         select "photoUrl"
+           from "vehicles"
+          where "vehicleId" = $5
+       ))
```

Comment:
You can simplify this quite a bit, since your form is being pre-populated now.

```sql
update "vehicles"
   set "year" = $1,
       "make = $2,
       "model" = $3,
       "photoUrl" = coalesce($4, "photoUrl")

-- etc
```